relay-server/src/utils/managed_envelope.rs

```diff
@@ -175,6 +213,16 @@ impl ManagedEnvelope {
         }
     }
 
+    /// Returns `true` if the indexed data category should be used for reporting.
+    ///
+    /// If metrics have been extracted from the event item, we use the indexed category
+    /// (for example, [TransactionIndexed](`DataCategory::TransactionIndexed`)) for reporting
+    /// rate limits and outcomes, because reporting of the main category will be handled by
+    /// the metrics aggregator.
```

Comment:
nit

> because reporting of the main category will be handled by the metrics aggregator.

I think "_main_ data category" may be a bit confusing, I'd add an example as with `TransactionIndexed`:

```suggestion
    /// If metrics have been extracted from the event item, we use the indexed category
    /// (for example, [TransactionIndexed](`DataCategory::TransactionIndexed`)) for reporting
    /// rate limits and outcomes, because reporting of the main category (for example, [Transaction]
    /// (`DataCategory::Transaction`) for processed transactions) will be handled by the metrics
    /// aggregator.
```